lib/LaTeXML/MathParser.pm

```diff
@@ -556,7 +556,13 @@ sub parse_kludge {
     my $kludge = $$pair[0];
     push(@replacements, (ref $kludge eq 'ARRAY') && ($$kludge[0] eq 'ltx:XMWrap')
       ? @$kludge[2 .. $#$kludge] : ($kludge)); }
-  $document->appendTree($mathnode, @replacements);
+  my $wrapped_replacements;
+  if (scalar(@replacements) == 1 and p_getQName($replacements[0]) eq 'ltx:XMWrap') {
+    $wrapped_replacements = $replacements[0];
+    p_setAttribute($wrapped_replacements, 'class', 'unparsed'); }
+  else {
```

Comment:
The `@class` is not needed, since the presence of `XMWrap` or `XMArg` already signifies that it's unparsed.

Reply:
I wanted to at least have the option of it reaching the HTML, so that formulas that failed to parse _could_ be styled differently, e.g. for web editing apps that use latexml behind the scenes and want all of their formulas to be parseable (and thus indexable).

Reply:
Hmm, that's getting ahead of yourself. But if so, then the class name doesn't follow our conventions.

Reply:
And, then probably should be integrated into post processing, so that pmml would add a class to the top-level, rather than redundantly carrying it around.